core/dbt/cli/flags.py

```diff
@@ -148,6 +152,10 @@ def assign_params(ctx, params_assigned_from_default, deprecated_env_vars):
                                 f"No deprecated param name match from {dep_name} to {new_name}"
                             )
 
+                        # remove from defaulted set
```

Comment:
Nit: Might be helpful to comment why we remove it from default(which is not obvious for me), the fact that we remove it  is relatively straight forwards to read